System/FilePath/Internal.hs

```diff
@@ -313,6 +313,15 @@ hasExtension :: FilePath -> Bool
 hasExtension = any isExtSeparator . takeFileName
 
 
+-- | Is the given string the final extension of the filename?
+--   The extension should not include the separator.
+--
+-- > "png" `isExtensionOf` "/directory/file.png"     == True
```

Comment:
@ndmitchell this is general question orthogonal to this PR: have you considered using doctest syntax? I.e.

```hs
-- >>> "png" `isExtensionOf` "/directory/file.png"
-- True
```

? I'm just asking, because we're slowly changing the markup in `base` to use that syntax, and I'm looking forward to cabal's upcoming native `cabal new-doctest` support...